futures-util/Cargo.toml

```diff
@@ -27,13 +27,15 @@ cfg-target-has-atomic = ["futures-core-preview/cfg-target-has-atomic"]
 sink = ["futures-sink-preview"]
 io = ["std", "futures-io-preview", "memchr"]
 channel = ["std", "futures-channel-preview"]
+join-macro = ["async-await", "futures-join-macro-preview", "proc-macro-hack", "proc-macro-nested"]
```

Comment:
For almost the same reason as `select!`, this is optional in `futures_util` (see https://github.com/rust-lang-nursery/futures-rs/issues/1701).